lib/samples/apply_symbology_to_shapefile/apply_symbology_to_shapefile.dart

```diff
@@ -0,0 +1,150 @@
+    with SampleStateSupport {
+  // Create a controller for the map view.
+  final _mapViewController = ArcGISMapView.createController();
+  // A flag for when the map view is ready and controls can be used.
+  var _ready = false;
+  // Hold reference to the feature layer so that its renderer can be changed when button is pushed.
+  late FeatureLayer _shapefileFeatureLayer;
+  // Hold reference to default renderer to enable switching back.
+  Renderer? _defaultRenderer;
+  // Hold reference to alternate renderer to enable switching.
+  SimpleRenderer? _alternateRenderer;
+  // Create variable for holding state relating to the renderer.
+  bool _alternate = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        top: false,
+        left: false,
+        right: false,
+        child: Stack(
+          children: [
+            Column(
+              children: [
+                Expanded(
+                  // Add a map view to the widget tree and set a controller.
+                  child: ArcGISMapView(
+                    controllerProvider: () => _mapViewController,
+                    onMapViewReady: onMapViewReady,
+                  ),
+                ),
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                  children: [
+                    const Text('Alternate renderer'),
+                    Switch(
+                      value: _alternate,
+                      onChanged: (value) {
+                        setState(() => _alternate = value);
+                        updateRenderer();
+                      },
+                    ),
+                  ],
+                ),
+              ],
+            ),
+            // Display a progress indicator and prevent interaction until state is ready.
+            LoadingIndicator(visible: !_ready),
+          ],
+        ),
+      ),
+    );
+  }
+
+  void onMapViewReady() async {
+    // Create a map with the topographic basemap style and an initial viewpoint.
+    final map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISTopographic);
+    map.initialViewpoint = Viewpoint.fromCenter(
+      ArcGISPoint(
+        x: -11662054,
+        y: 4818336,
+        spatialReference: SpatialReference(wkid: 3857),
+      ),
+      scale: 200000,
+    );
+
+    // Download the sample data.
+    await downloadSampleData(['d98b3e5293834c5f852f13c569930caa']);
+    // Get the application documents directory.
+    final appDir = await getApplicationDocumentsDirectory();
+    // Get the Shapefile from the download resource.
+    final shapefile =
+        File('${appDir.absolute.path}/Aurora_CO_shp/Subdivisions.shp');
+    // Create a feature table from the Shapefile URI.
+    final shapefileFeatureTable =
+        ShapefileFeatureTable.withFileUri(shapefile.uri);
+    // Create a feature layer for the Shapefile feature table.
+    setState(
```

Comment:
There's no need for `setState` here -- it should only be used when we're updating variables that change what the `build()` method returns.